security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -122,36 +109,38 @@ func (sc *SecretCache) rotate(t time.Time) {
 		proxyID := key.(string)
 		now := time.Now()
 
-		e := value.(SecretItem)
+		e := value.(sds.SecretItem)
 
 		// Remove from cache if the secret hasn't been used for a while, this prevent the cache growing indefinitely.
-		if now.After(e.lastUsedTime.Add(sc.evictionDuration)) {
+		if now.After(e.LastUsedTime.Add(sc.evictionDuration)) {
```

Comment:
Note, LastUsedTime should be modified in the fetch.

Reply:
it's modified in generateSecret(which fetch calls into).

Reply:
But we only set LastUsedTime when the secret is created? Do we need to have both CreationTime and LastUsedTime if they're all the same?

Reply:
sorry I forgot LastUsedTime was put here for fetchsecret's usage, which used to read from cache; since now we re-generated new key/cert for fetch/first stream call, LastUsedTime isn't needed anymore.